test/promql/tests/data_plane/performance_test.yaml

```diff
@@ -0,0 +1,280 @@
+      - series: 'kube_job_status_failed{namespace="default", job_name="example-job"}'
+        values: '1+0x15'
+
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: KubernetesJobSlowCompletion
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              job_name: example-job
+              namespace: default
+            exp_annotations:
+              summary: "Kubernetes Job slow completion."
+              description: "Kubernetes Job default/example-job did not complete in time."
+
+  - interval: 1m
+    input_series:
+      # Run status of 1 job is not available, so it will be alerted.
+      - series: 'kube_job_spec_completions{namespace="default", job_name="example-job"}'
+        values: '3+0x15'
+
+      # Status: 2 job succeeded
+      - series: 'kube_job_status_succeeded{namespace="default", job_name="example-job"}'
+        values: '2+0x15'
+
+      # Status: 0 jobs failed
+      - series: 'kube_job_status_failed{namespace="default", job_name="example-job"}'
+        values: '0+0x15'
+
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: KubernetesJobSlowCompletion
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              job_name: example-job
+              namespace: default
+            exp_annotations:
+              summary: "Kubernetes Job slow completion."
+              description: "Kubernetes Job default/example-job did not complete in time."
+
+  - interval: 1m
+    input_series:
+      # Run status of all jobs are available, so it will not be alerted.
+      - series: 'kube_job_spec_completions{namespace="default", job_name="example-job"}'
+        values: '3+0x15'
+
+      # Status: 3 job succeeded
+      - series: 'kube_job_status_succeeded{namespace="default", job_name="example-job"}'
+        values: '3+0x15'
+
+      # Status: 0 jobs failed
+      - series: 'kube_job_status_failed{namespace="default", job_name="example-job"}'
+        values: '0+0x15'
+
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: KubernetesJobSlowCompletion
+
+  - interval: 1m
+    input_series:
+      #  Run status of all jobs are available, so it will not be alerted.
+      - series: 'kube_job_spec_completions{namespace="default", job_name="example-job"}'
+        values: '3+0x15'
+
+      # Status: 2 job succeeded
+      - series: 'kube_job_status_succeeded{namespace="default", job_name="example-job"}'
+        values: '2+0x15'
+
+      # Status: 0 jobs failed
+      - series: 'kube_job_status_failed{namespace="default", job_name="example-job"}'
+        values: '1+0x15'
+
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: KubernetesJobSlowCompletion
+
+  # Node based Alerts
+  - interval: 1m
+    input_series:
+      # Node is running with 95.275% CPU usage, so it will be alerted.
+      - series: 'node_cpu_seconds_total{instance="instance1", mode="idle"}'
+        values: '0.05-0.1x10 8.5'
```

Comment:
This looks bit complicated. Is there a reason why not to have just `0.01x10` or so?

https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/

Reply:
I was actually trying to simulate here the test case for just above 95% usage, the alert should be triggered. I think we can also keep it simple. Let me revert it back.

Reply:
Makes sense, whatever works!

Reply:
I've updated the test case with 100% usage scenario.